pkg/didcomm/protocol/legacyconnection/models.go

```diff
@@ -76,3 +83,76 @@ type Connection struct {
 	DID    string   `json:"DID,omitempty"`
 	DIDDoc *did.Doc `json:"DIDDoc,omitempty"`
 }
+
+type connectionRaw struct {
+	DID    string  `json:"DID,omitempty"`
+	DIDDoc *rawDoc `json:"DIDDoc,omitempty"`
+}
+
+type rawDoc struct {
+	Context              interface{}              `json:"@context,omitempty"`
+	ID                   string                   `json:"id,omitempty"`
+	AlsoKnownAs          []interface{}            `json:"alsoKnownAs,omitempty"`
+	VerificationMethod   []map[string]interface{} `json:"verificationMethod,omitempty"`
+	PublicKey            []map[string]interface{} `json:"publicKey,omitempty"`
+	Service              []map[string]interface{} `json:"service,omitempty"`
+	Authentication       []interface{}            `json:"authentication,omitempty"`
+	AssertionMethod      []interface{}            `json:"assertionMethod,omitempty"`
+	CapabilityDelegation []interface{}            `json:"capabilityDelegation,omitempty"`
+	CapabilityInvocation []interface{}            `json:"capabilityInvocation,omitempty"`
+	KeyAgreement         []interface{}            `json:"keyAgreement,omitempty"`
+	Created              *time.Time               `json:"created,omitempty"`
+	Updated              *time.Time               `json:"updated,omitempty"`
+	Proof                []interface{}            `json:"proof,omitempty"`
+}
+
+// JSONBytes converts Connection to json bytes.
+func (con *Connection) JSONBytes() ([]byte, error) {
+	legacyDoc, err := con.DIDDoc.ToLegacyRawDoc()
+	if err != nil {
+		return nil, fmt.Errorf("converting to Legacy Raw Doc failed: %w", err)
+	}
+
+	connDoc := &rawDoc{}
+
+	_ = mapstructure.Decode(legacyDoc, connDoc) //nolint: errcheck
+
+	conRaw := connectionRaw{
+		DID:    con.DID,
+		DIDDoc: connDoc,
+	}
+
+	byteConn, err := json.Marshal(conRaw)
+	if err != nil {
+		return nil, fmt.Errorf("JSON marshalling of connection raw failed: %w", err)
+	}
+
+	return byteConn, nil
+}
+
+// ParseConnection creates an instance of Connection by reading a JSON connection from bytes.
+func ParseConnection(data []byte) (*Connection, error) {
+	raw := &connectionRaw{}
```

Comment:
I think the term `raw` here is a bit confusing. Can we use a term `legacy` instead?
So that we can have a connection and a legacy connection, a diddoc and a legacy diddoc.
Also we can have the methods
- `func (con *Connection) toLegacyJSONBytes() ([]byte, error)` instead of just JSONBytes
- `func ParseLegacyJSONBytes(data []byte) (*Connection, error)` instead of ParseConnection.